File: Easy/Valid-Parentheses.js
function isValid(s) {
  if (!s) {
    return false;
  }

  let stack = [];
  let closeToOpen = {
    ")": "(",
    "]": "[",
    "}": "{",
  };

  for (let parenthes of s) {
    if (closeToOpen.hasOwnProperty(parenthes)) {
      if (stack.length !== 0 && stack.at(-1) === closeToOpen[parenthes]) {
        stack.pop();
      } else {
        return false;
      }
    } else {
      stack.push(parenthes);
    }
  }

  return stack.length === 0;
}

File: Rust/Stack/Valid_Parentheses.rs
pub struct Solution {}

impl Solution {
    pub fn is_valid(s: String) -> bool {
        let mut stack = Vec::new();

        for i in s.chars() {
            match i {
                '{' => stack.push('}'),
                '(' => stack.push(')'),
                '[' => stack.push(']'),
                '}' | ')' | ']' if Some(i) != stack.pop() => return false,
                _ => (),
            }
        }

        stack.is_empty()
    }
}